core/controllers/improvements.py

```diff
@@ -50,23 +79,13 @@ def get(self, exploration_id):
 
     @acl_decorators.can_edit_exploration
     def post(self, exploration_id):
-        task_entries = self.payload.get('task_entries', None)
-        if task_entries is None:
-            raise self.InvalidInputException('No task_entries provided')
+        task_entries = self.normalized_payload.get('task_entries')
         task_entries_to_put = []
         for task_entry in task_entries:
-            entity_version = task_entry.get('entity_version', None)
-            if entity_version is None:
-                raise self.InvalidInputException('No entity_version provided')
-            task_type = task_entry.get('task_type', None)
-            if task_type is None:
-                raise self.InvalidInputException('No task_type provided')
-            target_id = task_entry.get('target_id', None)
-            if target_id is None:
-                raise self.InvalidInputException('No target_id provided')
-            status = task_entry.get('status', None)
-            if status is None:
-                raise self.InvalidInputException('No status provided')
+            entity_version = task_entry.get('entity_version')
```

Comment:
Probably use [] accessors rather than .get() so that any missing key is noisily rejected, rather than replaced with a default of None.

Also, do/should we be converting dicts to domain objects as part of the normalization so that this ambiguity does not arise?

Reply:
Thank you @seanlip -- As some changes are requested by @vojtechjelinek in this [discussion](https://github.com/oppia/oppia/pull/13422/files#r684793687) I am planning to remove domain objects file and transfer its contents into payload_validator.
While migrating I will make sure to update schema_utils for normalization of domain objects.
Please confirm does in sounds good or do I have to make changes here in.

Reply:
Ah, I see. This seems reasonable with one caveat: in cases where you return dict, call this a "dict" rather than a domain object (i.e. a "task entry dict" or similar). Never use "X" to represent "X dict", since that leads to confusion. If a variable is a dict, always make sure that any names which reference it have "dict" in them.

Could you please update the PR accordingly? Then I'd be happy to take a pass.

Thanks!

Reply:
Updated, please take a look, thank you.

Reply:
Thank you for review, will take care of these points while creating a seperate PR.